src/keep_gpu/utilities/rocm_visibility.py

```
@@ -0,0 +1,100 @@
+    if len(set(values)) != len(values):
+        return ()
+    return values
+
+
+def rocm_monitor_device_count(rocm_smi) -> Optional[int]:
+    count_fn = getattr(rocm_smi, "rsmi_num_monitor_devices", None)
+    if count_fn is None:
+        return None
+    try:
+        count = int(count_fn())
+    except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)